src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -70,7 +70,7 @@ private static void BindProperty(PropertyInfo property, object propertyOwner, IC
 
         private static object BindType(Type type, object typeInstance, IConfiguration configuration)
         {
-            var configValue = configuration.Get(null);
+            var configValue = configuration[null];
```

Comment:
Is this intentional or should we change the parameter type to `IConfigurationSection` and make this `configurationSection.Value`? I don't want to keep pushing forward this hacky pattern but I guess if it is important to have this method be able to target an `IConfigurationRoot` then our plans to specialize the node interfaces are in trouble.
Cc @lodejard @HaoK